src/objective-c/examples/BUILD

```diff
@@ -15,55 +15,154 @@
 # limitations under the License.
 
 
+load("//src/objective-c:grpc_objc_internal_library.bzl", "local_objc_grpc_library")
 load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
-load(
-    "@com_github_grpc_grpc//bazel:objc_grpc_library.bzl",
-    "objc_grpc_library",
-)
+load("@build_bazel_rules_apple//apple:tvos.bzl", "tvos_application")
+load("@build_bazel_rules_apple//apple:watchos.bzl", "watchos_application", "watchos_extension")
 
 proto_library(
     name = "messages_proto",
-    srcs = ["BazelBuildSamples/messages.proto"],
-    visibility = ["//visibility:public"],
+    srcs = ["RemoteTestClient/messages.proto"],
+)
+
+proto_library(
+    name = "test_proto",
+    srcs = ["RemoteTestClient/test.proto"],
+    deps = [":messages_proto"],
 )
 
-objc_grpc_library(
+# use objc_grpc_library in bazel:objc_grpc_library.bzl when developing outside the repo
+local_objc_grpc_library(
```

Comment:
What is the benefit of implementing "local" as another rule rather than a parameter?

Reply:
1. This is not supposed to be used outside the grpc repository (because it refers to local paths).
2. We don't use this in Google3

So I figured we do not want to merge it with `objc_grpc_library`.